src/studio/src/designer/backend/Controllers/TextsController.cs

```diff
@@ -30,12 +32,40 @@ public class TextsController : ControllerBase
         /// <summary>
         /// Initializes a new instance of the <see cref="TextsController"/> class.
         /// </summary>
-        /// <param name="textsService">The languages service.</param>
+        /// <param name="textsService">The texts service.</param>
         public TextsController(ITextsService textsService)
         {
             _textsService = textsService;
         }
 
+        /// <summary>
+        /// Endpoint for editing text file for a specific language.
+        /// </summary>
+        /// <param name="org">Unique identifier of the organisation responsible for the app.</param>
+        /// <param name="repo">Application identifier which is unique within an organisation.</param>
+        /// <param name="languageCode">Language identifier specifying the text file to read.</param>
+        /// <param name="jsonText">New content from request body to be added to the text file.</param>
+        /// <remarks>If duplicates of keys are tried added the
+        /// deserialization to dictionary will overwrite the first
+        /// key:value pair with the last key:value pair</remarks>
+        [HttpPut]
+        [Produces("application/json")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [Route("{languageCode}")]
+        public async Task<ActionResult> EditText(string org, string repo, string languageCode, [FromBody] Dictionary<string, string> jsonText)
```

Comment:
Ref. discussion on naming - UpdateText(s), Update or just Put (the context is the texts controller so texts is a bit redundant).